10up/ElasticPress
# Use must instead of should for 'or' relations

### Description of the Change
This PR will change the condition from using should to must for 'or' relations in meta queries. 


Closes #2932 

### How to test the Change

- While adding the query to sort the posts using key, it was taking the key inside the should clause, but now it should be under the must clause
- example: 
``` [
  'ep_integrate' => true,
  'post_type' => 'event',
  'meta_key' => 'event_start',
  'meta_query' => [
    'relation' =>'OR',
    [
      'key' => 'event_start',
      'value' => date('Ymd'),
      'compare' => '>=',
      'type' => 'NUMERIC',
    ],
    [
      'key' => 'event_end',
      'value' => date('Ymd'),
      'compare' => '>=',
      'type' => 'NUMERIC',
    ],
  ],
  'orderby' => 'meta_value_num',
  'order' => 'ASC',
] 
```
- Observe that the key is inside the must clause instead of should clause

### Changelog Entry

> Changed - The clause for meta_query when sorting by key and using 'or' relation


### Credits

Props @MARQAS 

### Checklist:
